tests/Controller/TaxonApiTest.php

```diff
@@ -0,0 +1,469 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Tests\Controller;
+
+use Lakion\ApiTestCase\JsonApiTestCase;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class TaxonApiTest extends JsonApiTestCase
+{
+    /**
+     * @var array
+     */
+    private static $authorizedHeaderWithContentType = [
+        'HTTP_Authorization' => 'Bearer SampleTokenNjZkNjY2MDEwMTAzMDkxMGE0OTlhYzU3NzYyMTE0ZGQ3ODcyMDAwM2EwMDZjNDI5NDlhMDdlMQ',
+        'CONTENT_TYPE' => 'application/json',
+    ];
+
+    /**
+     * @var array
+     */
+    private static $authorizedHeaderWithAccept = [
+        'HTTP_Authorization' => 'Bearer SampleTokenNjZkNjY2MDEwMTAzMDkxMGE0OTlhYzU3NzYyMTE0ZGQ3ODcyMDAwM2EwMDZjNDI5NDlhMDdlMQ',
+        'ACCEPT' => 'application/json',
+    ];
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_taxon_list_when_access_is_denied()
+    {
+        $this->loadFixturesFromFile('resources/taxons.yml');
+        $this->client->request('GET', '/api/v1/taxons/');
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_taxon_when_it_does_not_exist()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('GET', '/api/v1/taxons/-1', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_taxons()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/taxons.yml');
+
+        $this->client->request('GET', '/api/v1/taxons/', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'taxon/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_taxon()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $taxons = $this->loadFixturesFromFile('resources/taxons.yml');
+        $taxon = $taxons['women'];
+
+        $this->client->request('GET', '/api/v1/taxons/'.$taxon->getId(), [], [], static::$authorizedHeaderWithAccept);
```

Comment:
If we extract `'/api/v1/taxons/'.$taxon->getId()` to separate method it would be easier to change the way how we handle resource resolvation (e.g. `code` instead of `id`)